pkg/cli/proxy_get.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/k8s"
+)
+
+func GetEnvoyProxyConfig(clientSet kubernetes.Interface, config *rest.Config, namespace string, podName string, localPort uint16, query string) (string, error) {
+	// Check if the pod belongs to a mesh
+	pod, err := clientSet.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Errorf("Could not find pod %s in namespace %s", podName, namespace)
+	}
+	if !isMeshedPod(*pod) {
+		return "", errors.Errorf("Pod %s in namespace %s is not a part of a mesh", podName, namespace)
+	}
+	if pod.Status.Phase != corev1.PodRunning {
+		return "", errors.Errorf("Pod %s in namespace %s is not running", podName, namespace)
+	}
+
+	dialer, err := k8s.DialerToPod(config, clientSet, podName, namespace)
+	if err != nil {
+		return "", err
+	}
+
+	portForwarder, err := k8s.NewPortForwarder(dialer, fmt.Sprintf("%d:%d", localPort, constants.EnvoyAdminPort))
+	if err != nil {
+		return "", errors.Errorf("Error setting up port forwarding: %s", err)
+	}
+
+	var envoyProxyConfig string
+	err = portForwarder.Start(func(pf *k8s.PortForwarder) (error) {
+		defer pf.Stop()
+		url := fmt.Sprintf("http://localhost:%d/%s", localPort, query)
+
+		// #nosec G107: Potential HTTP request made with variable url
+		resp, err := http.Get(url)
+		if err != nil {
+			return errors.Errorf("Error fetching url %s: %s", url, err)
+		}
+
+		output, err := ioutil.ReadAll(resp.Body);
+		if err != nil {
+			return errors.Errorf("Error rendering HTTP response: %s", err)
+		}
+		envoyProxyConfig = string(output)
+		return nil
+	})
+	if err != nil {
+		return "", errors.Errorf("Error retrieving proxy config for pod %s in namespace %s: %s", podName, namespace, err)
+	}
+
+	return envoyProxyConfig, nil
+}
+
+// isMeshedPod returns a boolean indicating if the pod is part of a mesh
+func isMeshedPod(pod corev1.Pod) bool {
+	// osm-controller adds a unique label to each pod that belongs to a mesh
+	_, proxyLabelSet := pod.Labels[constants.EnvoyUniqueIDLabelName]
```

Comment:
`proxyLabelSet` is `ok` or simply a check for existence.  I'd call this `proxyLabelExists`

The `_` would be the value.  We seem to be ignoring it. I'd capture this value and check whether it is some valid UUId.

Reply:
@draychev How do we know if the uuid is valid? You mean we just need to check `_` and see if it _is a uuid?_